R/setup_codingclub_session.R

```diff
@@ -0,0 +1,148 @@
+#' setup_codingclub_session("20200225",
+#'   root_dir = "../",
+#'   src_rel_path = "source",
+#'   data_rel_path = "dataframes"
+#' )
+#' }
+#'
+#' @export
+
+setup_codingclub_session <- function(
+                                     session_date = format(Sys.Date(), "%Y%m%d"),
+                                     root_dir = ".",
+                                     src_rel_path = "src",
+                                     data_rel_path = "data") {
+  msg_session_date <- "session_date must be a string representing a date (YYYYMMDD)"
+  assert_that(is.string(session_date), msg = msg_session_date)
+  assert_that(grepl(
+    "\\d{4}(0[1-9]|1[012])(0[1-9]|[12][0-9]|3[01])",
+    session_date
+  ),
+  msg = msg_session_date
+  )
+
+  github_src_link <- sprintf(
+    "https://github.com/inbo/coding-club/tree/master/data/%s",
+    session_date
+  )
+  github_data_link <- sprintf(
+    "https://github.com/inbo/coding-club/tree/master/src/%s",
+    session_date
+  )
+  src_target_dir <- paste(root_dir, src_rel_path, session_date, sep = "/")
+  data_target_dir <- paste(root_dir, data_rel_path, session_date, sep = "/")
+  message(sprintf(
+    "R scripts in %s will be downloaded in folder: %s",
+    github_src_link, src_target_dir
+  ))
+  message(sprintf(
+    "Data in %s will be downloaded in folder: %s",
+    github_data_link, data_target_dir
+  ))
+  continue <- tolower(readline("Do you want to continue? (Y/N) "))
+  if (continue == "y") {
+    message("* Download source file(s)...")
+    content_downloaded <- download_content_in_subdir(session_date,
+      target_directory = src_target_dir,
+      github_subdirectory = "src"
+    )
+    if (content_downloaded) message("* Download source file(s) completed")
+
+    message("* Download data file(s)...")
+    content_downloaded <- download_content_in_subdir(session_date,
+      target_directory = data_target_dir,
+      github_subdirectory = "data"
+    )
+    if (content_downloaded) message("* Download data file(s) completed")
+  } else {
+    print("* Download aborted")
+  }
+}
+
+download_content_in_subdir <- function(session_date,
+                                       target_directory,
+                                       github_subdirectory) {
+  assert_that(github_subdirectory %in% c("src", "data"),
+    msg = "R scripts are in GitHub subfolder ./src/, data in ./data/"
+  )
+  content_found <- FALSE
+  content <- tryCatch(
+    {
+      resp <- gh(sprintf(
+        "/repos/inbo/coding-club/contents/%s/%s",
+        github_subdirectory, session_date
+      ))
+      content_found <- TRUE
+      resp
+    },
+    error = function(error_condition) {
+      warning(sprintf(
+        "No %s files found for session %s. Is the date correct?",
+        github_subdirectory, session_date
+      ))
+    }
+  )
+
+  if (content_found) {
+    if (!dir.exists(target_directory)) {
+      dir.create(target_directory, recursive = TRUE, showWarnings = TRUE)
```

Comment:
I tend to use `dir.create(showWarnings = FALSE)`. Then the dir is created if it doesn't exists. Otherwise nothing happens. This would remove the if-else structure. Making it easier to create a unit test.